util/kvdb-rocksdb/src/lib.rs

```diff
@@ -510,17 +512,25 @@ impl Database {
 
 	/// Get database iterator for flushed data.
 	pub fn iter(&self, col: Option<u32>) -> Option<DatabaseIterator> {
-		//TODO: iterate over overlay
 		match *self.db.read() {
 			Some(DBAndColumns { ref db, ref cfs }) => {
+				let overlay = &self.overlay.read()[Self::to_overlay_column(col)];
+				let overlay_data = overlay.iter()
```

Comment:
hashmap iterators aren't ordered, so interleave-ordered behavior probably won't be right there.